### PR TITLE
generalise support for cice filenames

### LINKED ISSUE #21: Update concat_ice_daily to support more cice variable filenames
We need to fix this line 

https://github.com/COSIMA/om3-scripts/blob/efb2167a1ecf8ef8355c9fa964f01319679e5bab/payu_config/archive_scripts/concat_ice_daily.sh#L54

to allow more [variety](https://github.com/COSIMA/access-om3/issues/191) of filenames:

(e.g. access-om3.cice.1day.????-??-01.nc )

and we should also check that the "output" filename is not overwriting an existing file.

### PR DESCRIPTION
**Pull request by anton-seaice**
Per https://github.com/COSIMA/access-om3/issues/190#issuecomment-2261753828 allow for support of more variety in filenames  and don't overwrite existing files

Closes #21

### CODE REVIEW

File: test/test_payu_conf/test_concat_ice_daily.py [hunk 1]
```diff
@@ -31,12 +30,8 @@ def daily_files(request, tmp_path):
 
     """
 
-    if request.param[0] == "Default":
+    if dir == "Default":
```
Comment (anton-seaice) [inline test/test_payu_conf/test_concat_ice_daily.py:33]:
```suggestion
    if dir_name == "Default":
```

**Review by minghangli-uni — COMMENTED** [review]
Thanks @anton-seaice. I've left some comments regarding the assumption that the 28th day is considered complete.

File: test/test_payu_conf/test_concat_ice_daily.py [hunk 2]
```diff
@@ -109,30 +112,58 @@ def test_true_case(daily_files, use_dir, nmonths, tmp_path):
...
-    for p in daily_files:
+    for p in daily_paths:
         assert_f_not_exists(p)
 
 
-@pytest.mark.parametrize(
-    "daily_files", [("Default", 1), ("Default", 27)], indirect=True
-)
-def test_no_concat_case(daily_files, tmp_path):
+@pytest.mark.parametrize("hist_dir, ndays", [("Default", 1), ("Default", 27)])
```
Comment (anton-seaice) [inline test/test_payu_conf/test_concat_ice_daily.py:119]:
```suggestion
@pytest.mark.parametrize("hist_dir, ndays", [("Default", 1), ("Default", 30)])
```

**Comment by anton-seaice** [comment]
The tests pass on gadi and fail on the github runner !

**Comment by anton-seaice** [comment]
I think this is ready to go @minghangli-uni

**Review by minghangli-uni — APPROVED** [review]
Looks great to me! Thanks @anton-seaice

**Pull request merged**